### DISCUSSION TITLE
What about Petzi?

### DISCUSSION THREAD

**Discussion by MarkJaroski** [jakopako/croncert-config#160] [Q&A] [answered]
They don't seem to have a terms of service forbidding scraping. Is it fair game?

https://www.petzi.ch/

**Comment by jakopako** [comment, marked as answer]
Generally, yes why not. However, my first concern would be duplicates. I haven't really properly tackled this yet and don't know if there is a simple solution due to inconsistent naming.

**Reply by MarkJaroski** [reply]
Yeah, duplicates would be the problem there. Also, I think some Petzi members don't list their free concerts.

**Comment by tits4net** [comment]
Hey, I'm from PETZI technical workgroup :)

I just found this discussion as we were discussing about data sharing. I comment to just let you know we are open for discussion on data sharing (and we have better interfaces then scrapping) ... even if it won't solve the duplicate issue ;-)

**Comment by MarkJaroski** [comment]
Wow. That's wonderful!

For what it's worth I've tried to make contact a few times, but I guess whoever reads the email filters this stuff out before it gets to you.

In any case I'd love to collaborate directly.

**Comment by tits4net** [comment]
Yep there is a chance that kind of request may be filtered at different level, sorry for that.  

For now, we can only provide "private" event endpoint, but on the year roadmap we have public event endpoint based on OLE (https://www.hinto.net/en/arbeiten-mit-hinto/ole-widget/ole-in-detail-70.html) that will be implemented. 

Do you see or would you prefer another option ? Do you want to wait for public endpoint or private one could already be interesting ?

**Comment by MarkJaroski** [comment]
I'm certainly willing to explore both options, presuming @jakopako is. My main concern would be to avoid duplicating events, but I have a couple of ideas about that.

Goskyr has the ability to filter, so for those Petzi members who do not list their free events, we can just filter out any ticketed events and get them from Petzi instead. In general this is going to lead to a higher quality and more stable feed.

Further down the line I guess we could try to convince the members to list their free events too.

Just coaxing people away from Instagram is already a chore.

Anyway, yeah! Let's see what we can do in collaboration!

**Reply by tits4net** [reply]
> Goskyr has the ability to filter, so for those Petzi members who do not list their free events, we can just filter out any ticketed events and get them from Petzi instead. In general this is going to lead to a higher quality and more stable feed.

Maybe I can give a few hints on this. Some PETZI member do not use the agenda at all, some use it only as an agenda and some use the ticketing system. So not sure if you will not have mismatch / duplicate but it's definitively a start. Maybe filtering by date could be an option, it's very uncommon to have multiple show the same date (even if it can happens in festivals)

> Further down the line I guess we could try to convince the members to list their free events too. 

We try since ages ... but some help is always welcome :)

**Reply by MarkJaroski** [reply]
It's like herding cats! Or worse.

I've reached out to the people who run Decadanse in Geneva about interlinking, and made the point that what they have there is kind of magical, so I don't want to touch it, just link to and from it.

**Comment by jakopako** [comment]
Hey @tits4net thanks for jumping into this discussion! @MarkJaroski I agree we should definitely explore both options. I guess we'll need to figure out how to convert between the different event formats but that should be feasible. I'm not sure if we should extend goskyr or just write a new tool/script for this. Maybe the latter would be easier.

The duplicates issue can probably be tackled with filters, as you're suggesting @MarkJaroski As long as just one source (any event website, petzi api... ) is 'responsible' for a venue it should be easy.

Looking forward to any kind of collaboration!

**Comment by MarkJaroski** [comment]
OLE is a semantic meta format inside of HTML, and we already use it in
several places! Occasionally I think goskyr even picks it up automatically
on generate mode.

So we wouldn't need a separate tool for that.

**Reply by jakopako** [reply]
Oooh haha good to know!
